libsycl/src/CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+      add_custom_command(
+        TARGET ${LIB_NAME} POST_BUILD
+        COMMAND ${CMAKE_COMMAND} -E copy
+        ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${LIB_NAME}.lib ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${ARG_IMPLIB_NAME}.lib
+        COMMENT "Creating version-agnostic copy of the import library.")
+      install(
+        FILES ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${ARG_IMPLIB_NAME}.lib
+        DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT libsycl)
+    endif()
+  endif()
+
+  if (MSVC)
+    # Install stripped PDB
+    check_linker_flag(CXX "LINKER:/PDBSTRIPPED:${LIB_NAME}.stripped.pdb"
+                          LINKER_SUPPORTS_PDBSTRIPPED)
+    if(LINKER_SUPPORTS_PDBSTRIPPED)
+      target_link_options(${LIB_NAME}
+                          PRIVATE "LINKER:/PDBSTRIPPED:${LIB_NAME}.stripped.pdb")
+      install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIB_NAME}.stripped.pdb"
+              DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
+              RENAME "${LIB_NAME}.pdb"
+              COMPONENT ${LIB_NAME}
+              OPTIONAL)
+    endif()
+  else()
+    target_compile_options(${LIB_OBJ_NAME} PUBLIC
+                           -fvisibility=hidden -fvisibility-inlines-hidden)
+
+    # Sycl math built-in macros cause a GCC 4.6 'note' to be output repeatedly.
+    # => note: the ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
+    # Seems to be no way to suppress it except use -Wno-psabi
+    target_compile_options(${LIB_OBJ_NAME} PUBLIC -Wno-psabi)
+
+    if (UNIX AND NOT APPLE)
+      set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/ld-version-script.txt")
+      target_link_libraries(
+          ${LIB_NAME} PRIVATE "-Wl,--version-script=${linker_script}")
+      set_target_properties(${LIB_NAME} PROPERTIES LINK_DEPENDS ${linker_script})
+    endif()
+  endif()
+
+  find_package(Threads REQUIRED)
+
+  target_link_libraries(${LIB_NAME}
+    PRIVATE
+      ${CMAKE_DL_LIBS}
+      ${CMAKE_THREAD_LIBS_INIT}
+  )
+
+  # Workaround for a bug in GCC version 5 and higher.
+  # More information https://bugs.launchpad.net/ubuntu/+source/gcc-5/+bug/1568899
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+      CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0)
+    target_link_libraries(${ARGV0} PRIVATE gcc_s gcc)
+  endif()
+
+  set_target_properties(${LIB_NAME} PROPERTIES
+                        VERSION ${LIBSYCL_VERSION_STRING}
+                        SOVERSION ${LIBSYCL_VERSION_STRING})
+
+  check_cxx_compiler_flag(-Winstantiation-after-specialization
+    HAS_INST_AFTER_SPEC)
+  if (HAS_INST_AFTER_SPEC)
+    target_compile_options(${LIB_OBJ_NAME} PRIVATE
+      -Winstantiation-after-specialization)
+  endif()
+endfunction(add_sycl_rt_library)
+
+set(LIBSYCL_SOURCES
+    "platform.cpp"
+)
+
+if (MSVC)
+  # MSVC provides two incompatible build variants for its CRT: release and debug
+  # To avoid potential issues in user code we also need to provide two kinds
+  # of SYCL Runtime Library for release and debug configurations.
```

Comment:
The comment says "two" but the code says "four" and covers the release/debug and threadsafe-static/dynamic.

Regardless, this is odd; I don't think the values of variables like `CMAKE_CXX_FLAGS_DEBUG` and `CMAKE_CXX_FLAGS_RELEASE` should be modified.

Reply:
I found that this WA may be not needed starting cmake 3.15 so I'll try to update and see if it helps.
Then probably I can remove this completely

reference: https://github.com/protocolbuffers/protobuf/blob/2d4414f384dc499af113b5991ce3eaa9df6dd931/CMakeLists.txt#L213

Reply:
I found that llvm project compiler-rt uses it too https://github.com/llvm/llvm-project/blob/main/compiler-rt/CMakeLists.txt#L421
They switched to the new approach with CMAKE_MSVC_RUNTIME_LIBRARY usage but still do this variables update to ensure in absence of duplication of CRT choices.
I assume we can left it as it is then and stick to this approach instead of adding a new one and keeping this as backup (as it is done in compiler-rt).
Please let me know what you think. 

regarding comment: I don't really get the concern here. Could you please clarify? we mention release and debug CRT versions and clear all related keys, we didn't aim to focus on MT or MD options here since we erase them all.

Reply:
Thanks, @KseniyaTikhomirova, I might have misread the code originally. I see this code has now been removed, so any concerns I had are moot anyway.

I think we should have at least one of the following though; probably the first one, but I'm not sure.
```
set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded)
set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
```

The use in compiler-rt looks to be an anachronism. That code was added in [commit 7bc09a4](https://github.com/llvm/llvm-project/commit/7bc09a471fbc274d8632d1379d4134bec63fecc4), but not removed when https://github.com/microsoft/vcpkg/pull/34123 was resolved. But there looks to be considerable history, so I could be misreading it.

Reply:
Ah, thank you, that sounds perfect. I suggested we should set `CMAKE_MSVC_RUNTIME_LIBRARY` explicitly because I saw it set for some other libraries. I re-checked and that was for compiler-rt libraries which probably do have special requirements. Thank you, this comment can be resolved.